### PR TITLE
openPMD: add software name version

### PR DESCRIPTION
**Pull request by s-sajid-ali**
Add software name and version as attributes to openPMD outputs.

### CODE REVIEW

**Comment by s-sajid-ali** [comment]
Pinging @ax3l for review.

File: CMakeLists.txt [hunk 1]
```diff
@@ -305,6 +313,7 @@ get_source_version(ImpactX ${ImpactX_SOURCE_DIR})
 configure_file(
     ${ImpactX_SOURCE_DIR}/src/ImpactXVersion.H.in
     ${ImpactX_BINARY_DIR}/src/ImpactXVersion.H
+    IMMEDIATE
```
Comment (ax3l) [inline CMakeLists.txt]: What does this do? :)

Have never seen it: https://cmake.org/cmake/help/latest/command/configure_file.html

File: CMakeLists.txt [hunk 2]
```diff
@@ -3,6 +3,14 @@
 cmake_minimum_required(VERSION 3.20.0)
 project(ImpactX VERSION 23.06)
 
+# generate compile_commands.json
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+# and place it in the source directory https://stackoverflow.com/a/60910583
+add_custom_target(
+  copy-compile-commands ALL
+  ${CMAKE_COMMAND} -E copy_if_different
+  ${CMAKE_BINARY_DIR}/compile_commands.json ${CMAKE_CURRENT_LIST_DIR})
```
Comment (ax3l) [inline CMakeLists.txt]: Why do you need the `compile_commands.json`?
Does your IDE not support CMake directly?

**Review by ax3l — COMMENTED** [review]
Thank you! :+1: 

Do you mind isolating the IDE-related changes and the I/O-related ones in two PRs? :)

File: CMakeLists.txt (hunk 1 above)
Comment (s-sajid-ali) [reply CMakeLists.txt]: Looks like it was a deprecated argument.

File: CMakeLists.txt (hunk 2 above)
Comment (s-sajid-ali) [reply CMakeLists.txt]: Maybe there are other ways, but the `compile_commands.json` is a rather convenient method : `https://clangd.llvm.org/config#compileflags`.

There are tools like [Bear](https://github.com/rizsotto/Bear) that can generate `compile_commands.json`, but generating it alongside the build reduces the number of tools needed.

**Comment by s-sajid-ali** [comment]
>Do you mind isolating the IDE-related changes and the I/O-related ones in two PRs? :)

Done: https://github.com/ECP-WarpX/impactx/pull/381

File: CMakeLists.txt (hunk 2 above)
Comment (s-sajid-ali) [reply CMakeLists.txt]: compile_commands.json is used by the C++ LSP (clangd) to [parse the source files](https://clangd.llvm.org/config#compileflags). Other build systems such as meson/bazel/etc [can also generate it](https://sarcasm.github.io/notes/dev/compilation-database.html#build-systems-and-compilers).

There are tools like [Bear](https://github.com/rizsotto/Bear) that can generate compile_commands.json, but generating it alongside the build reduces the number of tools needed.

Comment (s-sajid-ali) [reply CMakeLists.txt]: compile_commands.json is used by the C++ LSP (clangd) to [parse the source files](https://clangd.llvm.org/design/compile-commands). Other build systems such as meson/bazel/etc [can also generate it](https://sarcasm.github.io/notes/dev/compilation-database.html#build-systems-and-compilers), so it not really `CMake` specific.

There are tools like [Bear](https://github.com/rizsotto/Bear) that can generate compile_commands.json, but generating it alongside the build reduces the number of tools needed.

File: src/particles/elements/diagnostics/openPMD.cpp [hunk 3]
```diff
@@ -198,6 +199,7 @@ namespace detail
 #   endif
                 , "adios2.engine.usesteps = true"
             );
+            series.setSoftware("impactx", IMPACTX_VERSION);
```
Comment (ax3l) [inline src/particles/elements/diagnostics/openPMD.cpp]: For excellent spelling :)
```suggestion
            series.setSoftware("ImpactX", IMPACTX_VERSION);
```

**Review by ax3l — APPROVED** [review]
Thank you!

Please note the updated spelling I introduced in the last commit :)

**Pull request merged**